Operations/Extras/AddressBook/iOS/AddressBookOperations.swift

```diff
@@ -484,124 +484,5 @@ public class AddressBookDisplayNewPersonViewController<F: PresentingViewControll
-        AddressBookObserverQueue.queue.addOperation(addressBookObserverGroup!)
-    }
-
-    private mutating func stop() {
-        addressBookObserverGroup?.cancel()
-    }
-}
-
-extension AddressBookObserverQueue: OperationObserver {
-    public func operationDidStart(operation: Operation) {
-        print("Started listening for AddressBook changes.")
-    }
-
-    public func operation(operation: Operation, didProduceOperation newOperation: NSOperation) {
-        print("AddressBookObserver produced new Observer?")
-    }
-
-    public func operationDidFinish(operation: Operation, errors: [ErrorType]) {
-        print("Stopped listening for AddressBook changes.")
-    }
-}
-
-public class AddressBookObserverGroup: GroupOperation {
-
-    public typealias DidChangeBlock = (info: [NSObject: AnyObject]?) -> Void
-
-    class Observer: Operation {
-        typealias AddressBookDidChange = [NSObject: AnyObject]? -> Void
-
-        let addressBook: AddressBook
-        let addressBookDidChange: AddressBookDidChange
-        var observer: AddressBookExternalChangeObserver?
-
-        init(addressBook: AddressBook, block: AddressBookDidChange) {
-            self.addressBook = addressBook
-            self.addressBookDidChange = block
-            super.init()
-            addObserver(BackgroundObserver())
-        }
-
-        override func execute() {
-            observer = addressBook.observeExternalChanges { info in
-                self.addressBookDidChange(info)
-                self.finish(nil)
-            }
-        }
-    }
-
-    let didChangeBlock: DidChangeBlock
-    let accessAddressBook: AddressBookOperation
-
-    var addressBook: AddressBook? = .None
-    var observer: Observer? = .None
-
-    init(block: DidChangeBlock) {
-        didChangeBlock = block
-        accessAddressBook = AddressBookOperation()
-        super.init(operations: [ accessAddressBook ])
-        addCondition(MutuallyExclusive<AddressBookObserverGroup>())
-        addCondition(SilentCondition(AddressBookCondition()))
-    }
-
-    func addExternalChangeObserver() {
-        if !cancelled, let addressBook = addressBook {
-            let observer = Observer(addressBook: addressBook, block: didChangeBlock)
-            addOperation(observer)
-            self.observer = observer
-        }
-    }
-
-    public override func cancel() {
-        observer?.cancel()
-        super.cancel()
-    }
-
-    func addressBookDidChange(info: [NSObject: AnyObject]?) {
-        didChangeBlock(info: info)
-    }
-
-    public override func operationDidFinish(operation: NSOperation, withErrors errors: [ErrorType]) {
-        if errors.isEmpty {
-            if accessAddressBook == operation {
-                addressBook = accessAddressBook.addressBook
-                addExternalChangeObserver()
-            }
-            else if let _ = operation as? Observer {
-                addExternalChangeObserver()
-            }
-        }
-    }
-}
-
```

Comment:
I just deleted this external callback stuff, as it doesn't work, and it's a bit of a pain.